WcaOnRails/app/models/competition.rb

```diff
@@ -462,10 +462,14 @@ def end_date
   private def events_must_be_valid
     invalid_events = events - Event.all_official - Event.all_deprecated
     unless invalid_events.empty?
-      errors.add(:eventSpecs, "invalid event ids: #{invalid_events.map(&:id).join(',')}")
+      errors.add(:events, "invalid event ids: #{invalid_events.map(&:id).join(',')}")
     end
   end
 
+  def event_picker_events
+    competition_events.map(&:event_object).sort_by(&:rank)
+  end
```

Comment:
`event_object is unnecessary as commented in registration.rb. Also `event_picker_events` is unnecessary because it's the same (but an array) as `competitiom.events`.